stubs/protobuf/google/protobuf/internal/well_known_types.pyi

```diff
@@ -63,29 +67,33 @@ class _FieldMaskTree:
     def AddLeafNodes(self, prefix: tAny, node: tAny) -> None: ...
     def MergeMessage(self, source: tAny, destination: tAny, replace_message: tAny, replace_repeated: tAny) -> None: ...
 
+_StructValue: TypeAlias = struct_pb2.Struct | struct_pb2.ListValue | str | float | bool | None
+_StructValueArg: TypeAlias = _StructValue | Mapping[str, _StructValueArg] | Sequence[_StructValueArg]
+
 class Struct:
-    def __getitem__(self, key: tAny): ...
-    def __contains__(self, item: tAny): ...
-    def __setitem__(self, key: tAny, value: tAny) -> None: ...
-    def __delitem__(self, key: tAny) -> None: ...
+    def __getitem__(self, key: str) -> _StructValue: ...
+    def __contains__(self, item: _StructValueArg) -> bool: ...
+    def __setitem__(self, key: str, value: _StructValueArg) -> None: ...
+    def __delitem__(self, key: str) -> None: ...
     def __len__(self) -> int: ...
-    def __iter__(self): ...
-    def keys(self): ...
-    def values(self): ...
-    def items(self): ...
-    def get_or_create_list(self, key: tAny): ...
-    def get_or_create_struct(self, key: tAny): ...
-    def update(self, dictionary: tAny) -> None: ...
+    def __iter__(self) -> Iterator[str]: ...
+    def keys(self) -> Iterator[str]: ...
+    def values(self) -> Iterator[_StructValue]: ...
+    def items(self) -> Iterator[tuple[str, _StructValue]]: ...
+    def get_or_create_list(self, key: str) -> struct_pb2.ListValue: ...
+    def get_or_create_struct(self, key: str) -> struct_pb2.Struct: ...
+    def update(self, dictionary: Mapping[str, _StructValueArg]) -> None: ...
```

Comment:
Could potentially use our `SupportsItems` protocol for this one:

https://github.com/python/typeshed/blob/1c940e89141e0c09eeebbce2679760ec03173beb/stdlib/_typeshed/__init__.pyi#L112-L113

Reply:
Are we expected to use those protocols for type annotation, or mostly for structural subtyping?

Reply:
Also, not sure how an attempt to derive from that protocol would work with the stubtest.

Reply:
Not suggesting you should derive from the protocol. Just suggesting that the argument type doesn't necessarily need to be a `Mapping`, it only needs to have a `.items()` method, so you can just do:

```suggestion
    def update(self, dictionary: SupportsItems[str, _StructValueArg]) -> None: ...
```

Reply:
Ahh!